lang/cs/Org.Apache.REEF.Driver/StateMachine/SystemState.cs

```diff
@@ -0,0 +1,97 @@
+        {
+            { new StateTransition<SystemStates, SystemStateEvents>(SystemStates.WaitingForEvaluator, SystemStateEvents.AllContextsAreReady), SystemStates.SubmitingTasks },
+            { new StateTransition<SystemStates, SystemStateEvents>(SystemStates.WaitingForEvaluator, SystemStateEvents.NotRecoverable), SystemStates.Fail },
+            { new StateTransition<SystemStates, SystemStateEvents>(SystemStates.SubmitingTasks, SystemStateEvents.AllTasksAreRunning), SystemStates.TasksRunning },
+            { new StateTransition<SystemStates, SystemStateEvents>(SystemStates.SubmitingTasks, SystemStateEvents.FailedEvaluator), SystemStates.ShutingDown },
+            { new StateTransition<SystemStates, SystemStateEvents>(SystemStates.SubmitingTasks, SystemStateEvents.FailedTask), SystemStates.ShutingDown },
+            { new StateTransition<SystemStates, SystemStateEvents>(SystemStates.TasksRunning, SystemStateEvents.FailedEvaluator), SystemStates.ShutingDown },
+            { new StateTransition<SystemStates, SystemStateEvents>(SystemStates.TasksRunning, SystemStateEvents.FailedTask), SystemStates.ShutingDown },
+            { new StateTransition<SystemStates, SystemStateEvents>(SystemStates.TasksRunning, SystemStateEvents.AllTasksAreCompleted), SystemStates.TasksCompleted },
+            { new StateTransition<SystemStates, SystemStateEvents>(SystemStates.ShutingDown, SystemStateEvents.Recover), SystemStates.WaitingForEvaluator },
+            { new StateTransition<SystemStates, SystemStateEvents>(SystemStates.ShutingDown, SystemStateEvents.NotRecoverable), SystemStates.Fail }
+        };
+
+        private readonly object _stateLock = new object();
+        private SystemStates _currentState;
+
+        /// <summary>
+        /// create a new SysteState starting from WaitingForEvaluator
+        /// </summary>
+        public SystemState()
+        {
+            _currentState = SystemStates.WaitingForEvaluator;
+        }
+
+        /// <summary>
+        /// Returns current state
+        /// </summary>
+        public SystemStates CurrentState
+        {
+            get
+            {
+                lock (_stateLock)
+                {
+                    return _currentState;
+                }
+            }
+        }
+
+        /// <summary>
+        /// BAsed on the event and current state to determine the next state
+        /// </summary>
+        /// <param name="systemEvent"></param>
+        /// <returns></returns>
+        public SystemStates GetNext(SystemStateEvents systemEvent)
+        {
+            var transition = new StateTransition<SystemStates, SystemStateEvents>(_currentState, systemEvent);
+            SystemStates nextState;
+            if (!Transitions.TryGetValue(transition, out nextState))
+            {
+                throw new Exception("Invalid transition: " + _currentState + " -> " + systemEvent);
```

Comment:
I added an exception class in REEF-1327. Will use that one once REEF-1327 is merged.